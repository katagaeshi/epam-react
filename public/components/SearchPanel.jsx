import React from 'react';
import SearchField from './SearchField';
import SearchFilter from './SearchFilter';
import Button from './Button';
import '../styles/SearchPanel.css';

const labelText = 'FIND YOUR MOVIE';
const SearchPanel = () => (
  <div className="SearchPanel">
    <span className="Header">
      {labelText}
    </span>
    <SearchField />
    <SearchFilter />
    <Button text="SEARCH" />
  </div>
);

export default SearchPanel;
