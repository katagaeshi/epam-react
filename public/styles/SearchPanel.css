.SearchPanel {
  display: grid;
  grid-template-rows: 20px 20px 20px;
  grid-template-columns: 300px;
  background-color: gray;
  color: white;
}

.Header {
  grid-row: 1;
}

.SearchButtonContainer {
  grid-row: 3;
  grid-column: 2 / 2;
  text-align: right;
}

.SearchButton {
  background-color: pink;
  color: white;
  border: none;
  width: 100px;
  height: 20px;
}
