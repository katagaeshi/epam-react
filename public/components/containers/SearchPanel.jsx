import { connect } from 'react-redux';
import SearchPanel from '../ui/SearchPanel';
import {
  findMovies,
  setSearchText,
  setSearchBy,
  setRedirect,
  setCameFromLink,
} from '../../actions';

const mapStateToProps = state => Object.assign(
  {},
  state.searchPanel, {
    sortOption: state.sortBy,
    fetchFilmsOnLoad: state.foundMovies,
  },
);

const mapDispatchToProps = dispatch => ({
  startSearch(value, filter, sortOption) {
    if (!value) {
      console.log('Search field is empty');
      return;
    }
    dispatch(findMovies({
      value,
      filter,
      sortOption,
    }));
    dispatch(setRedirect(true));
    dispatch(setCameFromLink(false));
  },
  findMovies(query) {
    dispatch(findMovies({ query }));
  },
  handleSearchChange(event) {
    dispatch(setSearchText(event.target.value));
  },
  onFilterUpdate(changeEvent) {
    dispatch(setSearchBy(changeEvent.target.value));
  },
  setCameFromLink: value => dispatch(setCameFromLink(value)),
});

const Container = connect(mapStateToProps, mapDispatchToProps)(SearchPanel);
export default Container;
