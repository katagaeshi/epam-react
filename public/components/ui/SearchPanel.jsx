import React from 'react';
import PropTypes from 'prop-types';
import SearchField from './SearchField';
import SearchFilter from './SearchFilter';
import '../../styles/SearchPanel.css';

const labelText = 'FIND YOUR MOVIE';

const FILTER = {
  TITLE: 'title',
  GENRE: 'genres',
};

class SearchPanel extends React.Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    this.props.startSearch(
      this.props.text,
      FILTER[this.props.filter],
      this.props.sortOption,
    );
  }

  render() {
    return (
      <div className="SearchPanel">
        <span className="Header">
          {labelText}
        </span>
        <SearchField
          handleChange={this.props.handleSearchChange}
          onEnter={this.onClick}
        />
        <SearchFilter
          onUpdate={this.props.onFilterUpdate}
          checked={this.props.filter}
        />
        <button
          className="SearchButton"
          onClick={this.onClick}
        >
        SEARCH
        </button>
      </div>
    );
  }
}

SearchPanel.propTypes = {
  startSearch: PropTypes.func.isRequired,
  handleSearchChange: PropTypes.func.isRequired,
  onFilterUpdate: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  filter: PropTypes.string.isRequired,
  sortOption: PropTypes.string.isRequired,
};

export default SearchPanel;
