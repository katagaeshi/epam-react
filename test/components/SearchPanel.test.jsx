import React from 'react';
import renderer from 'react-test-renderer';
import SearchPanel from './../../public/components/SearchPanel';

describe('SearchPanel', () => {
  it('Snapshot', () => {
    const tree = renderer
      .create(<SearchPanel />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
