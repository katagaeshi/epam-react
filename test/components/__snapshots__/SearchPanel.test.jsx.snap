// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel Snapshot 1`] = `
<div
  className="SearchPanel"
>
  <span
    className="Header"
  >
    FIND YOUR MOVIE
  </span>
  <input
    className="SearchField"
    placeholder="Enter movie info here"
  />
  <div
    className="SearchFilter"
  >
    <label
      htmlFor="SearchFilter"
    >
      SEARCH BY
      <div
        className="RadioButton"
      >
        <input
          checked={true}
          id="SearchFilter-0"
          onChange={[Function]}
          type="radio"
          value="TITLE"
        />
        <label
          htmlFor="SearchFilter-0"
        >
          TITLE
        </label>
      </div>
      <div
        className="RadioButton"
      >
        <input
          checked={false}
          id="SearchFilter-1"
          onChange={[Function]}
          type="radio"
          value="GENRE"
        />
        <label
          htmlFor="SearchFilter-1"
        >
          GENRE
        </label>
      </div>
    </label>
  </div>
  <button
    className="SearchButton"
  >
    SEARCH
  </button>
</div>
`;
